client/consensus/slots/src/lib.rs

```diff
@@ -78,7 +78,12 @@ pub trait SlotWorker<B: BlockT> {
 	///
 	/// Returns a future that resolves to a [`SlotResult`] iff a block was successfully built in
 	/// the slot. Otherwise `None` is returned.
-	fn on_slot(&mut self, chain_head: B::Header, slot_info: SlotInfo) -> Self::OnSlot;
+	fn on_slot(
+		&mut self,
+		chain_head: B::Header,
+		slot_info: SlotInfo,
+		telemetry: Option<TelemetryHandle>,
```

Comment:
Instead of passing this here. The `SimpleSlotWorker` should get a function `telemetry(&self) -> Option<TelemetryHandle>`

Reply:
Done in 54390e7462